Broken in TypeScript 4.5.3

TypeScript 4.5.3
Platform: Windows 10
node: 17.9.0
node-dev: 7.4.1

```sh
PS C:\Users\hornta\Desktop\recipes\packages\scraper> node-dev ./src/index.ts
Debugger listening on ws://127.0.0.1:60616/76e837a4-c371-48d0-b358-b7da37bc5b8b
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
(node:71788) ExperimentalWarning: Custom ESM Loaders is an experimental feature. This feature could change at any time    
(Use `node --trace-warnings ...` to show where the warning was created)
TypeError [ERR_INVALID_RETURN_PROPERTY_VALUE]: Expected string, array buffer, or typed array to be returned for the "source" from the "load" function but got type object.
    at new NodeError (node:internal/errors:372:5)
    at assertBufferSource (node:internal/modules/esm/translators:82:9)
    at ESMLoader.moduleStrategy (node:internal/modules/esm/translators:113:3)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:361:14)
    at async link (node:internal/modules/esm/module_job:70:21)
Waiting for the debugger to disconnect...
[ERROR] 23:10:00 TypeError: Expected string, array buffer, or typed array to be returned for the "source" from the "load" 
function but got type object.
```

--- a/lib/loaders/get-format.mjs
+++ b/lib/loaders/get-format.mjs
@@ -13,6 +13,9 @@ export async function getFormat(url, context, defaultGetFormat) {
     return await defaultGetFormat(url, context, defaultGetFormat);
   } catch (error) {
     if (error.code !== 'ERR_UNKNOWN_FILE_EXTENSION') throw error;
-    return require('get-package-type')(required).then(format => ({ format }));
+    return require('get-package-type')(required).then(format => {
+      if (!['builtin', 'commonjs'].includes(format)) throw error;
+      return { format };
+    });
   }
 }

--- a/lib/loaders/load.mjs
+++ b/lib/loaders/load.mjs
@@ -13,6 +13,9 @@ export async function load(url, context, defaultLoad) {
     return await defaultLoad(url, context, defaultLoad);
   } catch (error) {
     if (error.code !== 'ERR_UNKNOWN_FILE_EXTENSION') throw error;
-    return require('get-package-type')(required).then(format => ({ format, source: null }));
+    return require('get-package-type')(required).then(format => {
+      if (!['builtin', 'commonjs'].includes(format)) throw error;
+      return { format };
+    });
   }
 }

--- a/package.json
+++ b/package.json
@@ -36,23 +36,23 @@
     "dynamic-dedupe": "^0.3.0",
     "filewatcher": "~3.0.0",
     "get-package-type": "^0.1.0",
-    "minimist": "^1.2.5",
+    "minimist": "^1.2.6",
     "node-notifier": "^8.0.1",
-    "resolve": "^1.0.0",
-    "semver": "^7.3.5"
+    "resolve": "^1.22.0",
+    "semver": "^7.3.7"
   },
   "devDependencies": {
-    "@types/node": "^16.11.3",
-    "eslint": "^8.10.0",
-    "eslint-plugin-import": "^2.22.1",
+    "@types/node": "^17.0.24",
+    "eslint": "^8.13.0",
+    "eslint-plugin-import": "^2.26.0",
     "husky": "^7.0.4",
-    "lint-staged": "^11.2.3",
-    "prettier": "^2.2.1",
-    "tap": "^15.1.6",
+    "lint-staged": "^12.3.8",
+    "prettier": "^2.6.2",
+    "tap": "^16.0.1",
     "tap-xunit": "^2.4.1",
     "touch": "^3.1.0",
-    "ts-node": "^10.3.1",
-    "typescript": "^4.1.5"
+    "ts-node": "^10.7.0",
+    "typescript": "^4.6.3"
   },
   "lint-staged": {
     "*.{js,mjs}": "eslint --cache --fix",

--- a/test/fixture/typescript-module/index.ts
+++ b/test/fixture/typescript-module/index.ts
@@ -1,0 +1,2 @@
+// We should not be able to load typescript files as ESModules
+export default ['test', 'data'] as const

--- a/test/fixture/typescript-module/package.json
+++ b/test/fixture/typescript-module/package.json
@@ -1,0 +1,1 @@
+{ "type": "module" }

--- a/test/spawn/typescript-module.js
+++ b/test/spawn/typescript-module.js
@@ -1,0 +1,14 @@
+const tap = require('tap');
+
+const { spawn } = require('../utils');
+
+tap.test(
+  'Gives ERR_UNKNOWN_FILE_EXTENSION when loading typescript files and package type is "module"',
+  t => {
+    spawn('typescript-module/index.ts', out => {
+      if (out.match(/ERR_UNKNOWN_FILE_EXTENSION/)) {
+        return { exit: t.end.bind(t) };
+      }
+    });
+  }
+);

